src/robusta/runner/web.py

```diff
@@ -57,11 +69,40 @@ def handle_alert_event():
         alert_manager_event = AlertManagerEvent(**req_json)
         for alert in alert_manager_event.alerts:
             alert = Web._relabel_alert(alert)
-            Web.alerts_queue.add_task(Web.event_handler.handle_trigger, PrometheusTriggerEvent(alert=alert))
+            alert_hash = Web.get_compound_hash([
+                alert.fingerprint.encode('ascii'),
+                alert.status.encode('utf-8'),
+                str(alert.startsAt.timestamp()).encode('ascii'),
+            ])
+            with Web.processed_alerts_cache_lock:
+                if alert_hash in Web.processed_alerts_cache:
+                    continue
+                else:
+                    Web.processed_alerts_cache[alert_hash] = True
+            Web.alerts_queue.add_task(
+                Web.event_handler.handle_trigger, PrometheusTriggerEvent(alert=alert)
+            )
 
         Web.event_handler.get_telemetry().last_alert_at = str(datetime.now())
         return jsonify(success=True)
 
+    @staticmethod
+    def processed_alerts_cache_flusher():
```

Comment:
I think we don't need this.
I think the TTL Cache handles expiration

Reply:
As discussed, this is implemented to handle some worst case scenarios when the built-in expiration mechanism in `TTLCache` would be run and take a very long time. To guard against this, we run `expire` periodically to distribute the workload.

Reply:
Okay, upon further inspection it looks like the "bulk expire", if it happens, in practice won't take so long to warrant having this mechanism implemented. Removed.